parsl/tests/test_mpi_apps/test_mpi_scheduler.py

```diff
@@ -218,3 +218,33 @@ def test_tiny_large_loop():
         got_result = scheduler.get_result(True, 1)
 
     assert got_result == result_pkl
+
+
+@pytest.mark.local
+def test_larger_jobs_prioritized():
+    """Larger jobs should be scheduled first"""
+
+    task_q, result_q = SpawnContext.Queue(), SpawnContext.Queue()
+    scheduler = MPITaskScheduler(task_q, result_q)
+
+    max_nodes = len(scheduler.available_nodes)
+
+    # The first task will get scheduled with all the nodes,
+    # and the remainder hits the backlog queue.
+    node_request_list = [max_nodes] + [random.randint(1, 4) for _i in range(8)]
+
+    for task_id, num_nodes in enumerate(node_request_list):
+        mock_task_buffer = pack_res_spec_apply_message("func", "args", "kwargs",
+                                                       resource_specification={
+                                                           "num_nodes": num_nodes,
+                                                           "ranks_per_node": 2
+                                                       })
+        task_package = {"task_id": task_id, "buffer": mock_task_buffer}
+        scheduler.put_task(task_package)
+
+    # Confirm that the tasks are sorted in decreasing order
```

Comment:
this test checks that tests coming out of the queue are sorted in priority order (that is, is the `priority` part of PrioritizedTask implemented correctly).

It doesn't check that they are sorted in num_nodes order, I think? (for example, introduce a bug into the calculation at  line 200 and I think this test will still pass, even though the code is buggy?)

Reply:
Good point, I'll update this test.

Reply:
I just pushed the updated test that tests for the backlog queue returning tasks in decreasing order of nodes requested.